client/ayon_core/plugins/publish/collect_rendered_files.py

```diff
@@ -73,30 +73,23 @@ def _process_path(self, data, anatomy):
         data_err = "invalid json file - missing data"
         required = ["user", "comment",
                     "job", "instances", "version"]
-        assert all(elem in data.keys() for elem in required), data_err
+
+        if any(elem not in data.keys() for elem in required):
```

Comment:
If `data` is a `dict` then this should work fine.
```suggestion
        if any(elem not in data for elem in required):
```

Reply:
true, but I think with `.keys()` it is much more explicit that you are working on dict by my opinion. But without keys it seems it is slightly faster :)